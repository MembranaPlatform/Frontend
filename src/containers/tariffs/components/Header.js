import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Header = ({ label, active, onClick }) => (
  <div className="tariffs__header-container">
    <div
      className={classNames('tariffs__header-label', {
        'active': active,
      })}
      onClick={onClick}
    >
      {label.toUpperCase()}
    </div>
  </div>
);

Header.defaultProps = {
  active: false,
};

Header.propTypes = {
  active: PropTypes.bool,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Header;
