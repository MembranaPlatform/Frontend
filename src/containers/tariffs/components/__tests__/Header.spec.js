import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('Header component matches snapshot', () => {
  const defaultProps = {
    label: 'Test label',
    active: false,
    selectable: false,
    onClick: jest.fn(),
  };

  it('with default props', () => {
    const wrapper = shallow(<Header {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('with `active` and `selectable` true', () => {
    const props = {
      ...defaultProps,
      active: true,
      selectable: true,
    };
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
