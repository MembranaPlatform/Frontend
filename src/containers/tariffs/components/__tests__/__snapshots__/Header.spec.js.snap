// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot with \`active\` and \`selectable\` true 1`] = `
<div
  className="tariffs__header-container"
>
  <div
    className="tariffs__header-label active selectable"
    onClick={[MockFunction]}
  >
    TEST LABEL
  </div>
</div>
`;

exports[`Header component matches snapshot with default props 1`] = `
<div
  className="tariffs__header-container"
>
  <div
    className="tariffs__header-label"
    onClick={[MockFunction]}
  >
    TEST LABEL
  </div>
</div>
`;
